proc_dash/utility.py

```diff
@@ -119,9 +120,14 @@ def are_subjects_same_across_pipelines(
     """Checks if subjects and sessions are the same across pipelines in the input."""
     pipelines_dict = extract_pipelines(bagel, schema)
 
-    pipeline_subject_sessions = [
-        df.loc[:, get_id_columns(bagel)] for df in pipelines_dict.values()
-    ]
+    pipeline_subject_sessions = []
+    for df in pipelines_dict.values():
+        # per pipeline, rows are sorted first in case participants/sessions are out of order
+        pipeline_subject_sessions.append(
+            df.sort_values(["participant_id", "session"]).loc[
```

Comment:
is `"participant_id", "session"` the same as ` get_id_columns(bagel)`? If so: better to stick with one. Otherwise code gets tricky to read. If not: Not sure what you're comparing over below.